File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n + 1];

        for (int i = 1; i < n + 1; i++) {
            int num = i;
            while (num != 0) {
                result[i] += num & 1;
                num = num >> 1;
            }
        }

        return result;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result = [0 for i in range(n + 1)]

        for i in range(1, n + 1):
            num = i
            while num:
                result[i] += num & 1
                num = num >> 1

        return result